webapp/src/client/client.js

```diff
@@ -263,62 +265,79 @@ export default class Client {
         };
     };
 
-    doGet = async (url, headers = {}, credentials) => {
-        headers.Accept = 'application/json';
-        let options = {
-            method: 'get',
-            headers,
-        };
+    
+    doGet = (url, headers = {}, credentials) => {
 
-        if (url.includes('plugins/' + pluginID)) {
-            options = Client4.getOptions(options);
-        }
+        return async (dispatch, getState) => {
+            const baseUrl = getPluginServerRoute(getState()) + "/"+ url;
+            headers.Accept = 'application/json';
+            let options = {
+                method: 'get',
+                headers,
+            };
 
-        if (credentials) {
-            options.credentials = credentials;
-        }
+            if (url.includes('plugins/' + pluginID)) {
+                options = Client4.getOptions(options);
```

Comment:
The prefix of the server URL should only occur on requests going to the Mattermost server. In this case all of the Mattermost requests include this plugin id structure. I think this makes sense:

```suggestion
                url = getPluginServerRoute(getState()) + "/"+ url;
                options = Client4.getOptions(options);
```

Then we can remove line 272